dashboard/src/actions/authActions.js

```diff
@@ -7,6 +7,49 @@ import Cookies from "js-cookie";
 import { showToast } from "actions/toastActions";
 import { uid } from "../utils/helper";
 
+
+// Create an Authentication Request
+export const authenticationRequest = () => async (dispatch, getState) => {
+    try {
+      const endpoints = getState().apiEndpoint.endpoints;
+      const oidcServer = endpoints["openid-connect"]?.issuer;
+      const oidcRealm = endpoints["openid-connect"]?.realm;
+      const oidcClient = endpoints["openid-connect"]?.client;
```

Comment:
I don't think `realm` is "optional", so handling a missing realm in an except is probably good. Isn't `client` still optional?.so quietly defaulting to `null` might not be a bad idea.

Reply:
My principal concern is that I think we over-use the [`?.` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) without giving proper thought to what it means.

If the referred-to property/mapping isn't present, then instead of getting an exception we get a value of `undefined`, which is fine unless we later try to use that value -- then we'll possibly incur a different error some distance away from where the problem actually originated, which will make the situation harder to reconcile.

In this case, do we really want to add `client_id=undefined` to the `GET` request?  Or, would we rather omit the `client_id` query parameter from the request, or just declare an error and skip making the request altogether?